internal/handler/user.go

```
@@ -0,0 +1,122 @@
+		return
+	}
+
+	user, err := userRepo.GetUserByID(r.Context(), id)
+	if err != nil {
+		logger.Error("failed to fetch user", "error", err)
+		http.Error(w, "user not found", http.StatusNotFound)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(user)
```

golangci-lint (1.62.2)

Error return value of `(*encoding/json.Encoder).Encode` is not checked

(errcheck)